File: client/src/lib/tradingUtils.ts
/**
 * Trading utility functions.
 * Centralizes common trading-related operations.
 */

/**
 * Normalize trading mode: 'live' -> 'real' for backend compatibility.
 * 
 * The frontend may use 'live' as a user-friendly term, but the backend
 * uses 'real' to represent live trading mode.
 * 
 * @param mode - Trading mode ('paper', 'real', or 'live')
 * @returns Normalized trading mode ('paper' or 'real')
 * 
 * @example
 * normalizeTradingMode("live") // returns "real"
 * normalizeTradingMode("real") // returns "real"
 * normalizeTradingMode("paper") // returns "paper"
 */
export function normalizeTradingMode(
  mode: "paper" | "real" | "live"
): "paper" | "real" {
  return mode === "live" ? "real" : mode;
}

File: server_fastapi/utils/trading_utils.py
"""
Trading utility functions.
Centralizes common trading-related operations.
"""


def normalize_trading_mode(mode: str) -> str:
    """
    Normalize trading mode: 'live' -> 'real' for backend compatibility.

    The frontend may use 'live' as a user-friendly term, but the backend
    uses 'real' to represent live trading mode.

    Args:
        mode: Trading mode string ('paper', 'real', or 'live')

    Returns:
        Normalized trading mode ('paper' or 'real')

    Example:
        >>> normalize_trading_mode("live")
        'real'

        >>> normalize_trading_mode("real")
        'real'

        >>> normalize_trading_mode("paper")
        'paper'
    """
    return "real" if mode == "live" else mode
